cdc/split.cc

```diff
@@ -70,13 +70,61 @@ struct partition_deletion {
     tombstone t;
 };
 
+using clustered_column_set = std::map<clustering_key, cdc::one_kind_column_set, clustering_key::less_compare>;
+
 struct batch {
     std::vector<static_row_update> static_updates;
     std::vector<clustered_row_insert> clustered_inserts;
     std::vector<clustered_row_update> clustered_updates;
     std::vector<clustered_row_deletion> clustered_row_deletions;
     std::vector<clustered_range_deletion> clustered_range_deletions;
     std::optional<partition_deletion> partition_deletions;
+
+    clustered_column_set get_affected_clustered_columns_per_row(const schema& s) const {
+        clustered_column_set ret{clustering_key::less_compare(s)};
+
+        if (!clustered_row_deletions.empty()) {
+            // When deleting a row, all columns are affected
+            cdc::one_kind_column_set all_columns{s.regular_columns_count()};
+            all_columns.set(0, s.regular_columns_count(), true);
+            for (const auto& change : clustered_row_deletions) {
+                ret.insert(std::make_pair(change.key, all_columns));
+            }
+        }
+
+        auto process_change_type = [&] (const auto& changes) {
+            for (const auto& change : changes) {
+                auto& cset = ret[change.key];
+                cset.resize(s.regular_columns_count());
+                for (const auto& entry : change.atomic_entries) {
+                    cset.set(entry.id);
+                }
+                for (const auto& entry : change.nonatomic_entries) {
+                    cset.set(entry.id);
+                }
+            }
+        };
+
+        process_change_type(clustered_inserts);
+        process_change_type(clustered_updates);
+
+        return ret;
+    }
+
+    cdc::one_kind_column_set get_affected_static_columns(const schema& s) const {
+        cdc::one_kind_column_set ret{s.static_columns_count()};
+
+        for (const auto& change : static_updates) {
+            for (const auto& entry : change.atomic_entries) {
```

Comment:
These two inner for loops are very similar and there are another two for loops like that in process_change_type. Maybe we could extract a helper function to avoid code duplication?

Reply:
Done.